src/filter/mod.rs

```diff
@@ -26,17 +26,21 @@ pub enum FilterError {
     /// Specific to a ProgrammableFilter, means no function is available as
     /// called in the script
     NoSuchFunction(&'static str, metric::Event),
+    /// Specific to a ProgrammableFilter, means the lua code errored
+    LuaError(String, metric::Event),
 }
 
-fn name_in_fe(fe: &FilterError) -> &'static str {
-    match *fe {
-        FilterError::NoSuchFunction(n, _) => n,
+fn msg_in_fe(fe: &FilterError) -> &str {
+    match fe {
+        &FilterError::NoSuchFunction(n, _) => n,
+        &FilterError::LuaError(ref n, _) => n,
     }
 }
 
 fn event_in_fe(fe: FilterError) -> metric::Event {
     match fe {
         FilterError::NoSuchFunction(_, m) => m,
```

Comment:
These arms could be combined, so 

```
FilterError::NoSuchFunction(_, m) | FilterError::LuaError(_, m) => m
```

Same with `msg_in_fe` above.

Reply:
compiler didn't like the `msg_in_fe` one:
```
  --> src/filter/mod.rs:35:73
   |
35 |         &FilterError::NoSuchFunction(n, _) | &FilterError::LuaError(ref n, _) => n,
   |                                      - first binding                    ^ bound in different ways

error[E0409]: variable `n` is bound in inconsistent ways within the same match arm
```
but done for event_in_fe.